src/object/srv_obj_migrate.c

```diff
@@ -493,8 +493,9 @@ migrate_pool_tls_create_one(void *data)
 
 	pool_child = ds_pool_child_lookup(arg->pool_uuid);
 	if (pool_child == NULL) {
-		D_ASSERTF(dss_get_module_info()->dmi_xs_id == 0,
-			  "Cannot find the pool "DF_UUIDF"\n", DP_UUID(arg->pool_uuid));
+		/* Local ds_pool_child isn't started yet, return a retry-able error */
+		if (dss_get_module_info()->dmi_xs_id != 0)
+			return -DER_STALE;
```

Comment:
What are the implications of returning -DER_STALE here in migrate? Will the IO be retried (as opposed to having the entire rebuild operation get a -DER_STALE result)? If the IO is retried, does this also force a pool map refresh task to be run first before the retry (and is that appropriate for this case)?  I'm not as familiar with the object IO code, but see obj_comp_cb() setting pm_stale = true and invoking obj_retry_cb() that adds a pool map refresh task.

Reply:
We've verified this fix by manual hotplug testing.

Reply:
OK I see, looks like the error would be propagated through such that the tgt scan handler would return that error to the leader.